template_test.go

```diff
@@ -23,100 +24,235 @@ import (
 )
 
 func TestGetTemplateByID(t *testing.T) {
-	responseJSONFile := "response/template_response.json"
-	responseData, err := ioutil.ReadFile(responseJSONFile)
-	if err != nil {
-		t.Fatalf("Failed to read response JSON file: %v", err)
+	tests := []struct {
```

Comment:
could you update this to match the format of the other tests; `map[string]struct`?
Having the name makes it easier to know what's being tested, imo.

Reply:
Fixed